pkg/control/sidecarcontrol/util.go

```diff
@@ -575,3 +626,11 @@ func IsSidecarContainer(container corev1.Container) bool {
 	}
 	return false
 }
+
+func IsPodSidecarSetUpgradeStateAwaitingPatchNormal(sidecarSet *appsv1alpha1.SidecarSet, pod *corev1.Pod) bool {
+	if pod.Annotations == nil || pod.Annotations[SidecarSetHashAnnotation] == "" {
+		return false
+	}
+	spec := GetPodSidecarSetUpgradeSpecInAnnotations(sidecarSet.Name, SidecarSetHashAnnotation, pod)
+	return spec.State == SidecarSetUpgradeStateUpdating || spec.State == ""
```

Comment:
I think we can disregard the stock scenario, so we can just judge the "spec.State == SidecarSetUpgradeStateUpdating"